### PR TITLE
feat(protocol): VersionMessage with (de)serialization

### PR DESCRIPTION
**Pull request by tdelabro**

### CODE REVIEW

**Review by AbdelStark — APPROVED** [review]
Lgtm

File: src/network/protocol.zig [hunk 1]
```diff
@@ -19,41 +25,152 @@ pub const COMMAND_SIZE: usize = 12;
...
+    /// The caller is responsible for freeing the returned value.
+    pub fn serialize(self: VersionMessage, allocator: std.mem.Allocator) ![]u8 {
+        // 4 + 8 + 8 + (2 * (8 + 16 + 2) + 8 + 4)
+        const fixed_length = 84;
+        var user_agent_len: usize = undefined;
+        if (self.user_agent) |ua| {
+            user_agent_len = ua.len;
+        } else {
+            user_agent_len = 0;
+        }
```
Comment (tcoratger) [inline src/network/protocol.zig:71]:
```suggestion
        const user_agent_len = if (self.user_agent) |ua| ua.len else 0;
```

File: src/network/protocol.zig [hunk 2]
```diff
@@ -19,41 +25,152 @@ pub const COMMAND_SIZE: usize = 12;
...
+
+    pub fn hintSerializedLen(self: VersionMessage) usize {
+        // 4 + 8 + 8 + (2 * (8 + 16 + 2) + 8 + 4)
+        const fixed_length = 84;
+        var user_agent_len: usize = undefined;
+        if (self.user_agent) |ua| {
+            user_agent_len = ua.len;
+        } else {
+            user_agent_len = 0;
+        }
```
Comment (tcoratger) [inline src/network/protocol.zig:154]: Here same comment as before

File: src/network/protocol.zig (hunk 1 above)
Comment (tdelabro) [reply src/network/protocol.zig:71]: Damn, I was looking for this exact syntax, but it is not showcased in the doc I found. Thank you

Comment (tdelabro) [reply src/network/protocol.zig]: done

File: src/network/protocol.zig (hunk 2 above)
Comment (tdelabro) [reply src/network/protocol.zig]: done

**Review by tcoratger — APPROVED** [review]

**Pull request merged**